frontend/main.go

```diff
@@ -145,16 +146,24 @@ func main() {
 		log.Fatalf("Error creating Kubernetes client: %s", err)
 	}
 
-	// Start server
-	r, err := startHTTPServer(&flags)
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
-	}
-
-	log.Println("Starting Odigos UI...")
-	log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
-	err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
+	// Attempt to start the server on port 3000 and increment the port number if it's already in use
+	for {
+		r, err := startHTTPServer(&flags)
+		if err != nil {
+			log.Printf("Error starting server on port %d: %s", flags.Port, err)
```

Comment:
Can you share an example of how to line will be printed to the user console? will the %s expand to long text or is it a short one?

Reply:
The line that's printed to the user console will look something like this if an error occurs:
` Error starting server on port 3000: <error_message_here> `

%d format specifier is for `flags.Port` integer variable and %s formatter is for the `err` string